test/integration/targets/file/tasks/main.yml

```diff
@@ -97,6 +97,17 @@
     that:
       - "file5_result.changed == true"
 
+- name: change soft link to relative
+  file: src={{output_file|basename}} dest={{output_dir}}/soft.txt state=link
+  register: file5a_result
+
+- name: verify that the file was marked as changed
+  assert:
+    that:
+      - "file5a_result.changed == true"
+      - "file5a_result.diff.before.src == output_file|expanduser"
+      - "file5a_result.diff.after.src == output_file|basename"
+
```

Comment:
If you could repeat the task and ensure `"file5a_result.changed == false"` that would be a good extra test